test/controllers/hosts_controller_test.rb

```diff
@@ -1310,6 +1310,26 @@ def test_submit_multiple_rebuild_config_pessimistic
     end
   end
 
+  test '#process_hostgroup works on Host subclasses' do
+    class Host::Test < Host::Base; end
+    hostgroup = FactoryGirl.create(:hostgroup)
+    host = FactoryGirl.create(:host, :type => "Host::Test", :hostgroup => hostgroup)
+    host.stubs(:set_hostgroup_defaults)
+    host.stubs(:set_compute_attributes)
+    host.stubs(:architecture)
+    host.stubs(:operatingsystem)
+    host.stubs(:environment)
+    host.stubs(:domain)
+    host.stubs(:subnet)
+    host.stubs(:compute_profile)
+    host.stubs(:realm)
+    attrs = host_attributes(host)
+    attrs[:id] = host.id
+    attrs[:hostgroup_id] = hostgroup.id
+    xhr :put, :process_hostgroup, { :host => attrs }, set_session_user
```

Comment:
`set_session_user` uses admin user so the authorization code is skipped, should it be tested under user to verify `.authorized(:view_hosts)` works correctly?